src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5149,7 +5149,27 @@ public override BoundNode VisitUnaryOperator(BoundUnaryOperator node)
         {
             Debug.Assert(!IsConditionalState);
 
-            _ = base.VisitUnaryOperator(node);
+            switch (node.OperatorKind)
```

Comment:
>switch (node.OperatorKind) [](start = 12, length = 26)

Should we move this more accurate analysis into the base class?

Reply:
The base class has the invariant that conditional states are only for `bool` values.  That is relaxed here.

---
In reply to: [276386205](https://github.com/dotnet/roslyn/pull/34996#discussion_r276386205) [](ancestors = 276386205)